src/test/kotlin/graphics/scenery/tests/examples/volumes/ClientApplication.kt

```diff
@@ -0,0 +1,509 @@
+import graphics.scenery.volumes.vdi.VDINode
+import net.imglib2.type.numeric.integer.UnsignedIntType
+import net.imglib2.type.numeric.real.FloatType
+import org.joml.Matrix4f
+import org.joml.Vector3f
+import org.joml.Vector3i
+import org.lwjgl.system.MemoryUtil
+import org.zeromq.SocketType
+import org.zeromq.ZContext
+import org.zeromq.ZMQ
+import org.zeromq.ZMQException
+import java.io.ByteArrayInputStream
+import java.nio.ByteBuffer
+import java.util.concurrent.atomic.AtomicInteger
+import kotlin.concurrent.thread
+import kotlin.system.measureNanoTime
+
+class ClientApplication : SceneryBase("Client Application", 512, 512)  {
+
+    var hmd: TrackedStereoGlasses? = null
+    var buffer: ByteBuffer = ByteBuffer.allocateDirect(0)
+    val context = ZContext(4)
+
+    val numSupersegments = 20
+    val skipEmpty = true
+    var vdiStreaming = true
+    var newVDI = false
+    var firstVDI = true
+    var firstVDIStream = true
+    var firstVR = true
+    var currentlyVolumeRendering = false
+    var videoDecoding = true
+
+    val compute = VDINode()
+    val switch = EmptyNode()
+
+    val vulkanProjectionFix =
+        Matrix4f(
+            1.0f, 0.0f, 0.0f, 0.0f,
+            0.0f, -1.0f, 0.0f, 0.0f,
+            0.0f, 0.0f, 0.5f, 0.0f,
+            0.0f, 0.0f, 0.5f, 1.0f)
+    fun Matrix4f.applyVulkanCoordinateSystem(): Matrix4f {
+        val m = Matrix4f(vulkanProjectionFix)
+        m.mul(this)
+        return m
+    }
+    override fun init() {
+
+        //Step 1: Create necessary common components
+        renderer = hub.add(Renderer.createRenderer(hub, applicationName, scene, windowWidth, windowHeight))
+
+        val cam: Camera = DetachedHeadCamera()
+        with(cam) {
+            name = "ClientCamera"
+            spatial {
+                position = Vector3f(0.0f, 0.0f, 5.0f)
+            }
+            perspectiveCamera(50.0f, 512, 512)
+            scene.addChild(this)
+        }
+
+        //Step 2: Create necessary video-streaming  components
+        val dummyVolume = DummyVolume()
+        with(dummyVolume) {
+            name = "DummyVolume"
+            transferFunction = TransferFunction.ramp(0.1f, 0.5f)
+            scene.addChild(this)
+        }
+
+        val VRPlane = FullscreenObject()
+        with(VRPlane){
+            name = "VRplane"
+            VRPlane.wantsSync = false
+        }
+
+        val bridge = SwingBridgeFrame("1DTransferFunctionEditor")
+        val tfUI = TransferFunctionEditor(dummyVolume)
+        bridge.addPanel(tfUI)
+        tfUI.name = dummyVolume.name
+        val swingUiNode = bridge.uiNode
+        swingUiNode.spatial() {
+            position = Vector3f(2f,0f,0f)
+        }
+
+        //Step 3: Create necessary vdi-streaming  components
+        val opBuffer = MemoryUtil.memCalloc(windowWidth * windowHeight * 4)
+        compute.name = "vdi node"
+        compute.setMaterial(ShaderMaterial(Shaders.ShadersFromFiles(arrayOf("VDIRenderer.comp"), this@ClientApplication::class.java)))
```

Comment:
As stated in the VDINode class, this is something that ought to be happening there.